lib/rocket_chat/messages/channel.rb

```diff
@@ -58,6 +58,27 @@ def online(room_id: nil, name: nil)
         response['online'].map { |hash| RocketChat::User.new hash } if response['success']
       end
 
+      #
+      # channels.members REST API
+      # @param [String] room_id Rocket.Chat room id
+      # @param [String] name Rocket.Chat room name
+      # @param [Integer] offset Query offset
+      # @param [Integer] count Query count/limit
+      # @param [Hash] sort Query field sort hash. eg `{ msgs: 1, name: -1 }`
+      # @param [Hash] fields Query fields to return. eg `{ name: 1, ro: 0 }`
+      # @param [Hash] query The query. `{ active: true, type: { '$in': ['name', 'general'] } }`
+      # @return [Room[]]
+      # @raise [HTTPError, StatusError]
+      #
+      def members(room_id: nil, name: nil, offset: nil, count: nil, sort: nil, fields: nil, query: {})
```

Comment:
I added `room_id` and `name` as first arguments for this method, I think wold be consistent with the other methods in this file.

This is a method which is also a list and the arguments `offset: nil, count: nil, sort: nil, fields: nil, query: {}` make sense.

I'm not sure about the `fields` looking at the [RocketChat API docs](https://docs.rocket.chat/api/rest-api/methods/channels/members) this is not supported, I added it for consistency. 

We could also omit it, and use `{}` when building the params for list `build_list_body(offset, count, sort, {}, query)`

I'm having doubts on what is the best approach, @abrom could you please advise when you have the chance?

Thank you 🙇

Reply:
Correct, the API would support both the room ID and name.

The API does not support either `fields` or `query` parameters so please remove these. We wouldn't want someone to think that because the method accepts these parameters that the API does too!

If you just pass through the `fields` value as `nil` it will not be included.

Given that the [groups.members](https://docs.rocket.chat/api/rest-api/methods/groups/members) API also supports the same parameters and returns the same response I would suggest you put this API code in the parent class `Room`. That way it will be available to both. As per the other methods in the `Room` class you'd pass the URI for the API through using the `api_path` helper: `self.class.api_path('members')`

Reply:
@abrom Thank you for the feedback 🙇‍♀️ I applied your suggestions.